[MinkhantEain] ip

MinkhantEain opened the pull request:
# Luke
> "I am your father" - [Darth Vader](https://en.wikipedia.org/wiki/Darth_Vader)

The name Luke was chosen for the following reasons:
+ Similar to Duke
- ~~Easy~~ Simple change from Duke
* Same first name as __Luke Skywalker__ :baby: 

What can it do for you?
1. It can track your list of things to do
2. It can show the deadline :nerd_face: 
3. It can show the task timeline :open_mouth: 

All this functionality at **SGD__0.00__!**

Features:
- [x] Basic task tracking functionality
- [ ] GUI implementation (Coming Soon)
- [ ] Push notification (Coming Soon)

Just for info, the `main` method looks like this:
```java
public static void main(String[] args) throws IOException {
        new Luke("./data/Luke.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

MinkhantEain committed:
Level 0. Rename, Greet, Exit

MinkhantEain committed:
Level 1. Echo

MinkhantEain committed:
Level 2. Add, list

MinkhantEain committed:
Task file created

MinkhantEain committed:
Todo.java added

MinkhantEain committed:
Deadline added

MinkhantEain committed:
Event added

MinkhantEain committed:
Level 4

MinkhantEain committed:
corrected detection of /by /from and /to

MinkhantEain committed:
test script implemented

MinkhantEain committed:
LukeException created

MinkhantEain committed:
Error handled and tests updated

MinkhantEain committed:
Delete feature added

MinkhantEain committed:
Refactored Tasks to factory pattern

MinkhantEain committed:
Add save

MinkhantEain committed:
Merge branch 'branch-Level-7'

MinkhantEain committed:
Add date formatting

MinkhantEain committed:
Merge branch 'branch-Level-8'

MinkhantEain committed:
Add Ui, Storage, Parser, TaskList

MinkhantEain committed:
Merge tag 'A-MoreOOP'

MinkhantEain committed:
Add packages

MinkhantEain committed:
Merge branch 'add-gradle-support'

MinkhantEain committed:
Add gradle configuration

MinkhantEain committed:
Add JUnit tests

MinkhantEain committed:
A-Jar

MinkhantEain committed:
Add find feature

MinkhantEain committed:
Add JavaDoc

MinkhantEain committed:
Standardised Coding format

MinkhantEain committed:
Conflict resolution

MinkhantEain committed:
Merge A-JavaDoc conflict resolution

MinkhantEain committed:
Fix bug in luke.java

MinkhantEain committed:
Add JavaFx

MinkhantEain committed:
Add task pages and navigation

MinkhantEain committed:
Add todo save

MinkhantEain committed:
Add add task and save functionality

MinkhantEain committed:
Add gui feature

MinkhantEain committed:
Add test

MinkhantEain committed:
Add chat box

MinkhantEain committed:
Resolve merge conflict

itsnotsherm commented on src/main/java/presentation/Ui.java:
```diff
@@ -0,0 +1,87 @@
+    /**
+     * Shows closing dialog when the application ends.
+     */
+    public void closingDialog() {
```
Would it be clearer to have your method be named "closeDialog" without the adverb?

itsnotsherm commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.Objects;
+
+public abstract class Task {
+    private boolean completed;
```
I like your naming. Should this boolean be named isComplete instead?

itsnotsherm commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,64 @@
+
+    public static Event load(String input) {
+        String[] parameters = input.split("\\|");
+        String[] dateRange = parameters[3].split("to");
```
Perhaps a better naming to include a plural noun instead so that it is clearly a collection of Strings?

itsnotsherm commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+        this.completed = false;
+    }
+
+    public String completedStringRepresentation() {
```
Should this be renamed to a verb that represents the method?

itsnotsherm commented on src/main/java/utility/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+    private final ArrayList<Task> tasks;
+
+    /** Returns the size of tasks. */
+    public int getTasksSize() {
```
Would it be clearer if this is named getTaskListSize instead?

itsnotsherm commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public int intComplete() {
+        return completed ? 1 : 0;
```
Any idea why you do this instead of "return completed == true"?

itsnotsherm reviewed (commented):
LGTM, just a few nits to fix with regards to variable and method names

MinkhantEain committed:
Correct style

MinkhantEain committed:
Refactor files

The codes are messy and the code quality was not there for some files.

It needs to be more readable.

Let's refactor the modified files to make it more readable.

By being more readable, it will be easier to read and easier for new
programmers to work on this project.

MinkhantEain committed:
Add assert to modified files

Assumptions in program is not addressed.

Better to state the assumption explicitly to crash the program
so that it will not corrput luke's data in program.

Let's add assertion in places where we have underlying assumption.

Using assert will crash the program.

MinkhantEain committed:
Add streams to modified files

No use of streams in current files.

Using streams for parts of the application can make
it easier for us to reason about and debug the program.

Let's change parts that is changable to streams into
streams.

This will make those parts less error prone and easier
to reason about.

MinkhantEain committed:
Merge Level-10 to master

MinkhantEain committed:
Add assert at Storage constructor

MinkhantEain committed:
Merge pull request #1 from MinkhantEain/branch-A-Assertions

Branch a assertions

MinkhantEain committed:
Merge branch 'branch-A-CodeQuality' into master

MinkhantEain committed:
Merge pull request #2 from MinkhantEain/master

branch-A-CodeQuality

MinkhantEain committed:
Change after merge

MinkhantEain committed:
Merge branch 'branch-A-Streams' into master

MinkhantEain committed:
Correct failing tests

MinkhantEain committed:
Add addNote functionality to note with test

No note taking functionality for the Luke app.

Having note taking feature will enrich user experience.

Let's incrementally add addNote function in order to deliver
this new feature.

Increment addition will make the process easier to maintain.

MinkhantEain committed:
Add note taking feature

There is no way for customer to take note.

Need note taking feature to make user's life better.

Add note taking feature.

Let's add note taking feature.

It is done this way.

MinkhantEain committed:
Add CI file.

No CI file.

Need to add CI file to automate test at git.

Add CI file to auto test, I'm not sure.

Following tuturial.

Let's add CI file into my work.

It is done this way to meet IP requirement.

MinkhantEain committed:
Check code style

MinkhantEain committed:
Merge branch 'branch-BCD-Extension'

MinkhantEain committed:
Fix out of bound error

MinkhantEain committed:
Merge pull request #4 from MinkhantEain/branch-A-CodeQuality

Merge pull request #2 from MinkhantEain/master

MinkhantEain committed:
Change image

MinkhantEain committed:
Integrate gui to Luke

MinkhantEain committed:
Complete gui integration to Luke

MinkhantEain committed:
Check style

MinkhantEain committed:
Change pic

MinkhantEain committed:
Add user guide

MinkhantEain committed:
Add release ver

MinkhantEain committed:
Update javadoc

MinkhantEain committed:
Fix failing test case and correct checkstyle
